src/main/java/com/microsoft/jenkins/acs/commands/AKSDeploymentCommand.java

```diff
@@ -48,10 +47,9 @@ public void execute(IAKSDeploymentCommandData context) {
 
         @Override
         protected void prepareKubeconfig(FilePath kubeconfigFile) throws Exception {
-            AzureCredentials.ServicePrincipal sp = getServicePrincipal();
-            Azure azureClient = AzureHelper.buildClientFromServicePrincipal(sp);
+            Azure azureClient = AzureHelper.buildClient(credentialId);
```

Comment:
Does this work on slave?

Previously the `ServicePrincipal` contains all the information required to build the authorized Azure client.